Mysql upgrade doesn't work

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh

I'm reporting a bug, not asking a question, but I can't know whether it's reproducible for others.

- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? 
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?

Sorta, the passenger and nginx keg-only formulae shouldn't affect mysql at all, and I'm not going to reformat my hd as case insensitive just for a bug report, but mysql should run just fine on case sensitive file systems, as that's the norm on linux, and looking at the error logs it doesn't look like making the fs case insensitive would improve things any.
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
```
brew gist-logs mysql
Error: Your macOS keychain GitHub credentials do not have sufficient scope!
Scopes they need: gist
Scopes they have: 
Create a personal access token: https://github.com/settings/tokens/new?scopes=gist,public_repo&description=Homebrew
and then set HOMEBREW_GITHUB_API_TOKEN as the authentication method instead.
/usr/local/Homebrew/Library/Homebrew/utils/github.rb:214:in `raise_api_error': GitHub::HTTPNotFoundError
	from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:180:in `open_api'
	from /usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:116:in `create_gist'
	from /usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:58:in `gistify_logs'
	from /usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:129:in `gist_logs'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:87:in `<main>'
```
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
```
$ brew config && brew doctor
HOMEBREW_VERSION: 1.7.1-34-gb335a73
ORIGIN: https://github.com/Homebrew/brew
HEAD: b335a7357b7c02603ee9a5e15a666563ce02654c
Last commit: 27 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 61f5dca8fd24567db3ce8c8ce1506a9c13301da6
Core tap last commit: 47 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_AUTO_UPDATE_SECS: 86400
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_INSTALL_BADGE: 🍷
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 9.1 build 902
Git: 2.18.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: N/A
macOS: 10.13.6-x86_64
CLT: 9.4.1.0.1.1528165917
Xcode: 9.4.1
XQuartz: N/A
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The filesystem on / appears to be case-sensitive.
The default macOS filesystem is case-insensitive. Please report any apparent problems.

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  nginx
  passenger
```
To help us debug your issue please explain:
- What you were trying to do (and why)

Trying to upgrade mysql from 8.0.11 to 8.0.12, because it's good to keep sw up to date

- What happened (include command output)

See logs below

- What you expected to happen

It's a minor mysql version bump, I expected it to work without intervention.

- Step-by-step reproduction instructions (by running `brew install` commands)

`brew update && brew upgrade`

Note: previous formulae removed for brevity
```
==> Upgrading mysql 
==> Downloading https://homebrew.bintray.com/bottles/mysql-8.0.12.high_sierra.bottle.tar.gz
######################################################################## 100.0%
==> Pouring mysql-8.0.12.high_sierra.bottle.tar.gz
==> /usr/local/Cellar/mysql/8.0.12/bin/mysqld --initialize-insecure --user=camdennarzt --basedir=/usr/local/Cellar/mysql/8.0.12 --datadir=/usr/local/var/mysql --tmpdir=/tmp
Last 15 lines from /Users/camdennarzt/Library/Logs/Homebrew/mysql/post_install.01.mysqld:
2018-07-28 10:07:37 -0600

/usr/local/Cellar/mysql/8.0.12/bin/mysqld
--initialize-insecure
--user=camdennarzt
--basedir=/usr/local/Cellar/mysql/8.0.12
--datadir=/usr/local/var/mysql
--tmpdir=/tmp

2018-07-28T16:07:37.968266Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2018-07-28T16:07:37.968603Z 0 [System] [MY-013169] [Server] /usr/local/Cellar/mysql/8.0.12/bin/mysqld (mysqld 8.0.12) initializing of server in progress as process 76537
2018-07-28T16:07:37.970348Z 0 [ERROR] [MY-010457] [Server] --initialize specified but the data directory has files in it. Aborting.
2018-07-28T16:07:37.970384Z 0 [ERROR] [MY-010119] [Server] Aborting
2018-07-28T16:07:37.970546Z 0 [System] [MY-010910] [Server] /usr/local/Cellar/mysql/8.0.12/bin/mysqld: Shutdown complete (mysqld 8.0.12)  Homebrew.
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall mysql`
==> Caveats
We've installed your MySQL database without a root password. To secure it run:
    mysql_secure_installation

MySQL is configured to only allow connections from localhost by default

To connect run:
    mysql -uroot

To restart mysql after an upgrade:
  brew services restart mysql
Or, if you don't want/need a background service you can just run:
  mysql.server start
==> Summary
🍷  /usr/local/Cellar/mysql/8.0.12: 255 files, 233.0MB
==> Caveats
==> mysql
We've installed your MySQL database without a root password. To secure it run:
    mysql_secure_installation

MySQL is configured to only allow connections from localhost by default

To connect run:
    mysql -uroot

To restart mysql after an upgrade:
  brew services restart mysql
Or, if you don't want/need a background service you can just run:
  mysql.server start
```

--- a/Formula/mysql.rb
+++ b/Formula/mysql.rb
@@ -120,7 +120,7 @@ class Mysql < Formula
   def post_install
     # Make sure the datadir exists
     datadir.mkpath
-    unless (datadir/"mysql/user.frm").exist?
+    unless (datadir/"mysql/general_log.CSM").exist?
       ENV["TMPDIR"] = nil
       system bin/"mysqld", "--initialize-insecure", "--user=#{ENV["USER"]}",
         "--basedir=#{prefix}", "--datadir=#{datadir}", "--tmpdir=/tmp"

--- a/Formula/mysql@5.5.rb
+++ b/Formula/mysql@5.5.rb
@@ -120,7 +120,7 @@ class MysqlAT55 < Formula
   def post_install
     # Make sure the datadir exists
     datadir.mkpath
-    unless (datadir/"mysql/user.frm").exist?
+    unless (datadir/"mysql/general_log.CSM").exist?
       ENV["TMPDIR"] = nil
       system bin/"mysql_install_db", "--verbose", "--user=#{ENV["USER"]}",
         "--basedir=#{prefix}", "--datadir=#{datadir}", "--tmpdir=/tmp"

--- a/Formula/mysql@5.6.rb
+++ b/Formula/mysql@5.6.rb
@@ -120,7 +120,7 @@ class MysqlAT56 < Formula
   def post_install
     # Make sure the datadir exists
     datadir.mkpath
-    unless (datadir/"mysql/user.frm").exist?
+    unless (datadir/"mysql/general_log.CSM").exist?
       ENV["TMPDIR"] = nil
       system bin/"mysql_install_db", "--verbose", "--user=#{ENV["USER"]}",
         "--basedir=#{prefix}", "--datadir=#{datadir}", "--tmpdir=/tmp"

--- a/Formula/mysql@5.7.rb
+++ b/Formula/mysql@5.7.rb
@@ -113,7 +113,7 @@ class MysqlAT57 < Formula
   def post_install
     # Make sure the datadir exists
     datadir.mkpath
-    unless (datadir/"mysql/user.frm").exist?
+    unless (datadir/"mysql/general_log.CSM").exist?
       ENV["TMPDIR"] = nil
       system bin/"mysqld", "--initialize-insecure", "--user=#{ENV["USER"]}",
         "--basedir=#{prefix}", "--datadir=#{datadir}", "--tmpdir=/tmp"

